### PR TITLE
bt-323: [BUG FIX] no redirect to 404 on wrong url

### LINKED ISSUE #323: No redirect to 404 page with wrong link in address bar
**Description:**
No redirect to the 404 page when putting a wrong link in the address bar from the onboarding steps.

**Steps to reproduce:**
1. Sign in to the app as a Talent
2. When you are on the onboarding steps page put any symbols at the end of the link in the address bar of your browser
3. Press enter

**Expected result:**
Redirected to 404 page if an address does not exist

**Actual result:**
First step Profile opened

**Visual proof:**
Image

**Severity:**
Minor

### PR DESCRIPTION
**Pull request by JosuerBague**
This fixes the previous behaviour where accessing invalid routes of `/talent/onboarding/steps/:step` would not render the not found page.

Issue was caused by :step params defaulting to 'profile' when :step was not a valid onboarding route.

These changes include a refactoring of the constants found in:
`frontend/src/bundles/talent-onboarding/constants/constants.ts`

Further Route issues (structure and protected routes) are being addressed in the following PR: https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/pull/309

### CODE REVIEW

File: frontend/src/bundles/talent-onboarding/constants/constants.ts
Comment (nikita-remeslov) [inline frontend/src/bundles/talent-onboarding/constants/constants.ts:1]: why do we need this constant?

Comment (nikita-remeslov) [inline frontend/src/bundles/talent-onboarding/constants/constants.ts:1]: https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/blob/development/frontend/src/index.tsx#L35-L79

also, these routes should be children of App.tsx.

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/talent-onboarding/constants/constants.ts
Comment (JosuerBague) [reply frontend/src/bundles/talent-onboarding/constants/constants.ts:1]: It's being used in finding what slug was used for navigating to this particular onboarding path. I did not implement it, just refactored a bit while experimenting with it when I was fixing the 404 not found.

Should I remove these changes?

**Comment by JosuerBague** [comment]
Redundant PR, bug was fixed in another ticket.

**Pull request closed without merging**